tests/preprocessing/test_positionfixes.py

```diff
@@ -1,116 +1,225 @@
 import datetime
 import os
 import sys
+from math import radians
 
 import geopandas as gpd
 import numpy as np
 import pandas as pd
-from sklearn.cluster import DBSCAN
+import pytest
 from shapely.geometry import Point
-from math import radians
+from sklearn.cluster import DBSCAN
 
 import trackintel as ti
 from trackintel.geogr.distances import haversine_dist
 
+
+@pytest.fixture
+def pfs_geolife():
+    return ti.io.dataset_reader.read_geolife(os.path.join('tests', 'data', 'geolife'))
+
+@pytest.fixture
+def pfs_geolife_long():
+    return ti.io.dataset_reader.read_geolife(os.path.join('tests', 'data', 'geolife_long'))
+
+
+@pytest.fixture
+def geolife_pfs_spts_short(pfs_geolife):
+    pfs, spts = pfs_geolife.as_positionfixes.generate_staypoints(method='sliding', dist_threshold=25,
+                                                                 time_threshold=5 * 60)
+    return pfs, spts
+
+
+@pytest.fixture
+def geolife_pfs_spts_long(pfs_geolife_long):
+    pfs, spts = pfs_geolife_long.as_positionfixes.generate_staypoints(method='sliding', dist_threshold=25,
+                                                                      time_threshold=5 * 60)
+    return pfs, spts
+
+
+@pytest.fixture
+def testdata_tpls_geolife():
+    tpls_file = os.path.join('tests', 'data', 'geolife', 'geolife_triplegs_short.csv')
+    tpls = ti.read_triplegs_csv(tpls_file, tz='utc', index_col='id')
+    tpls.geometry = tpls.geometry.set_crs(epsg=4326)
+    return tpls
+
+
+@pytest.fixture
+def testdata_pfs_ids_geolife():
+    pfs_file = os.path.join('tests', 'data', 'geolife', 'geolife_positionfixes_with_ids.csv')
+    pfs = ti.read_positionfixes_csv(pfs_file, index_col='id')
+    pfs.geometry = pfs.geometry.set_crs(epsg=4326)
+    return pfs
+
+
+def assert_geodataframe_equal(gdf1, gdf2, **kwargs):
```

Comment:
same as the fixture functions. Move to the end?

Reply:
👍